src/module.cpp

```diff
@@ -3029,10 +3092,19 @@ int Module::CompileAndOutput(const char *srcFile, Arch arch, const char *cpu, st
         lEmitDispatchModule(dispatchModule, exportedFunctions);
 
         if (outFileName != NULL) {
-            if ((outputType == Bitcode) || (outputType == BitcodeText))
+            switch (outputType) {
+            case CPPStub:
+                break;
+
+            case Bitcode:
+            case BitcodeText:
                 writeBitcode(dispatchModule, outFileName, outputType);
-            else
+                break;
+
+            default:
+                Assert((outputType == Module::Object || outputType == Module::Asm) && "Unexpected `outputType`");
```

Comment:
Let's have a case for `Object` and `Asm` and assert in default case that it should not be reachable.